### PR TITLE
Added searchPlaceholder and search img icon to the RCV view

### PR DESCRIPTION
**Pull request by zahnno**
Updated UI within the rich choice view. 

- Included a scope search image.
- Implemented a custom search placeholder.
- Hide header if none is present.
- Accounted for absence of header in styles.

### CODE REVIEW

File: src/foam/u2/view/RichChoiceView.js [hunk 1]
```diff
@@ -270,7 +291,7 @@ foam.CLASS({
                   .addClass('search')
                   .add(self.FILTER_.clone().copyFrom({ view: {
                     class: 'foam.u2.view.TextField',
-                    placeholder: 'Search...',
+                    placeholder: this.searchPlaceholder ? this.searchPlaceholder : 'Search...',
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]: Can just set the `value` of the `searchPlaceholder` property instead.

File: src/foam/u2/view/RichChoiceView.js [hunk 2]
```diff
@@ -286,7 +307,7 @@ foam.CLASS({
               var index = 0;
               return this.E().forEach(sections, function(section) {
                 this
-                  .start().hide(!! section.hideIfEmpty && resp[index].value <= 0)
+                  .start().hide(!! section.hideIfEmpty && resp[index].value <= 0 || ! section.heading)
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js:314]: Why do you want to hide sections that don't have headings?

**Review by TW80000 — CHANGES_REQUESTED** [review]
You added an image and added styles for the image but the image is never actually inserted into the view.

File: src/foam/u2/view/RichChoiceView.js (hunk 2 above)
Comment (TW80000) [reply src/foam/u2/view/RichChoiceView.js:314]: Oh, I see, you're just hiding the heading itself, not the whole section.

Comment (zahnno) [reply src/foam/u2/view/RichChoiceView.js:314]: It's not hiding the section, it's hiding the header is none is passed in. Also did this so the styles applied to self.myClass('heading') aren't applied into the view.

**Pull request merged**